src/blockdev.rs

```diff
@@ -42,6 +42,47 @@ use crate::util::*;
 
 use crate::{runcmd, runcmd_output};
 
+#[derive(Debug, Deserialize)]
+struct DevicesOutput {
+    blockdevices: Vec<Device>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Device {
+    pub name: String,
+    pub label: Option<String>,
+    pub fstype: Option<String>,
+    #[serde(rename = "type")]
+    pub blktype: Option<String>,
+    pub mountpoint: Option<String>,
+    pub uuid: Option<String>,
+    pub children: Option<Vec<Device>>,
+}
+
+impl Device {
+    pub fn lsblk(dev: &Path, with_children: bool) -> Result<Device> {
+        let mut cmd = Command::new("lsblk");
+        cmd.args(&[
+            "-J",
+            "--paths",
+            "-o",
+            "NAME,LABEL,FSTYPE,TYPE,MOUNTPOINT,UUID",
+        ])
+        .arg(dev);
+        if !with_children {
+            cmd.arg("--nodeps");
+        }
+        let output = cmd_output(&mut cmd)?;
+        let devs: DevicesOutput = serde_json::from_str(&output)?;
+        let devinfo = devs
+            .blockdevices
+            .into_iter()
+            .next()
+            .ok_or_else(|| anyhow!("failed to get device information"))?;
```

Comment:
This should verify that there is only one dev and return that dev. Otherwise error out.